app/src/main/java/me/saket/dank/ui/submission/SortingAndTimePeriod.java

```diff
@@ -35,6 +36,58 @@ public static SortingAndTimePeriod create(SubredditSort sortOrder, TimePeriod ti
     return new AutoValue_SortingAndTimePeriod(sortOrder, timePeriod);
   }
 
+  public static SortingAndTimePeriod create(String sortOrderTimePeriod) {
+    String[] splittedString = sortOrderTimePeriod.split(SEPARATOR);
```

Comment:
It might be a better idea to use `Moshi` for serializing and deserializing `SortingAndTimePeriod`. Doing it by hand is prone to errors and breakage in case the schema changes in the future.

Reply:
I need to read more about Moshi. I've tried to use it in another feature with no luck. 👍

Reply:
I'll gladly walk you through Moshi if you want